frontend/src/components/module-editor/SideBar.tsx

```diff
@@ -1,46 +1,126 @@
-import React from "react";
+import React, { useContext } from "react";
+import { useParams } from "react-router-dom";
 import {
   Box,
+  Button,
   Flex,
+  Link,
+  HStack,
   Tabs,
   TabList,
   Tab,
   TabPanels,
   TabPanel,
+  Text,
 } from "@chakra-ui/react";
+import { ChevronLeftIcon, EditIcon } from "@chakra-ui/icons";
 
 import ModuleOverview from "./SideBarModuleOverview";
 import ContentKiosk from "./SideBarContentKiosk";
+import { ReactComponent as SaveIcon } from "../../assets/Save.svg";
+import { ADMIN_DASHBOARD_PAGE } from "../../constants/Routes";
+import EditorContext from "../../contexts/ModuleEditorContext";
+import { ModuleEditorParams } from "../../types/ModuleEditorTypes";
 
 const Sidebar = (): React.ReactElement => {
+  const { moduleIndex }: ModuleEditorParams = useParams();
+
+  const context = useContext(EditorContext);
+
+  if (!context) return <></>;
+  const { state } = context;
   return (
-    <Box w="20%">
+    <Box w="20%" minW="300px">
       <Flex
         position="fixed"
-        top="0"
-        left="0"
-        p="4"
         w="inherit"
-        h="100%"
-        minW="min-content"
+        minW="300px"
```

Comment:
If the parent has a minimum width, this flex should simply receive it's width right? Or am I wrong on that? There might be something weird about using `min-content` when the child is of `position=fixed`. Anyway, if it works it works (but I'm still not a fan of the pixel value)